diff --git a/cc/private/rules_impl/cc_static_library.bzl b/cc/private/rules_impl/cc_static_library.bzl
--- a/cc/private/rules_impl/cc_static_library.bzl
+++ b/cc/private/rules_impl/cc_static_library.bzl
@@ -39,7 +39,6 @@ def _flatten_and_get_objects(linker_inputs):
     for linker_input in linker_inputs.to_list():
         for lib in linker_input.libraries:
             if lib._contains_objects:
-                transitive_objects.append(depset(lib.pic_objects))
-                transitive_objects.append(depset(lib.objects))
+                transitive_objects.append(depset(lib.pic_objects or lib.objects))

     return depset(transitive = transitive_objects, order = "topological")
