diff --git a/src/parser/cxx/type_checker.cc b/src/parser/cxx/type_checker.cc
--- a/src/parser/cxx/type_checker.cc
+++ b/src/parser/cxx/type_checker.cc
@@ -3110,8 +3110,14 @@

     ExpressionAST* initializer = ast->bracedInitList;
     ast->constructorSymbol = check.check_class_initializer(
         ast->type, initializer, ast->bracedInitList->lbraceLoc);
+    return;
+  }
+
+  if (unqualified_cast<VectorType>(ast->type)) {
+    check.check_braced_init_list(ast->type, ast->bracedInitList,
+                                 InitializationKind::kDirectListInitialization);
     return;
   }

   ast->bracedInitList->type = ast->type;
