diff --git a/src/parser/cxx/ast_rewriter_expressions.cc b/src/parser/cxx/ast_rewriter_expressions.cc
--- a/src/parser/cxx/ast_rewriter_expressions.cc
+++ b/src/parser/cxx/ast_rewriter_expressions.cc
@@ -602,6 +602,7 @@

 auto ASTRewriter::ExpressionVisitor::operator()(IdExpressionAST* ast)
     -> ExpressionAST* {
+  const auto isCallee = std::exchange(rewrite.rewritingCallee_, false);
   if (auto pack = rewrite.functionParameterPackFor(ast->symbol)) {
     if (auto expandedParam = rewrite.packElementAt(pack)) {
       auto copy = IdExpressionAST::create(arena());
@@ -646,10 +647,8 @@
       }
     }
   } else if (copy->nestedNameSpecifier && copy->nestedNameSpecifier->symbol) {
-    binder()->qualifiedLookupIdExpression(copy);
+    binder()->qualifiedLookupIdExpression(copy, isCallee);
   } else if (ast->symbol) {
-    const auto isCallee = std::exchange(rewrite.rewritingCallee_, false);
-
     copy->symbol = rewrite.remapSymbol(ast->symbol);

     if (!isCallee && symbol_cast<OverloadSetSymbol>(copy->symbol) &&
diff --git a/src/parser/cxx/binder.cc b/src/parser/cxx/binder.cc
--- a/src/parser/cxx/binder.cc
+++ b/src/parser/cxx/binder.cc
@@ -3521,7 +3521,7 @@
   resolveIdExpression(ast, mayUseArgumentDependentLookup);
 }

-void Binder::qualifiedLookupIdExpression(IdExpressionAST* ast) {
+void Binder::qualifiedLookupIdExpression(IdExpressionAST* ast, bool isCallee) {
   if (!ast->unqualifiedId) return;
   if (!ast->nestedNameSpecifier || !ast->nestedNameSpecifier->symbol) return;

@@ -3549,7 +3549,7 @@
     return;
   }

-  resolveIdExpression(ast, /*isCallee=*/false);
+  resolveIdExpression(ast, isCallee);

   if (auto function = designatedFunction(ast->symbol)) {
     ast->symbol = function;
diff --git a/src/parser/cxx/binder.h b/src/parser/cxx/binder.h
--- a/src/parser/cxx/binder.h
+++ b/src/parser/cxx/binder.h
@@ -386,7 +386,7 @@

   void resolveIdExpression(IdExpressionAST* ast, bool isCallee);

-  void qualifiedLookupIdExpression(IdExpressionAST* ast);
+  void qualifiedLookupIdExpression(IdExpressionAST* ast, bool isCallee);

   [[nodiscard]] auto resolve(NestedNameSpecifierAST* nestedNameSpecifier,
                              UnqualifiedIdAST* unqualifiedId,
