diff --git a/src/parser/cxx/ast_interpreter_expressions.cc b/src/parser/cxx/ast_interpreter_expressions.cc
--- a/src/parser/cxx/ast_interpreter_expressions.cc
+++ b/src/parser/cxx/ast_interpreter_expressions.cc
@@ -2036,7 +2036,8 @@

         if (sz <= 4) {
           if (unit()->typeTraits().is_unsigned(ast->expression->type)) {
-            return toValue(~toUInt32(expressionResult.value()));
+            return toValue(static_cast<std::uintmax_t>(
+                ~toUInt32(expressionResult.value())));
           }

           return ExpressionResult(
